src/Console/Command/FixCommand.php

```diff
@@ -302,9 +302,11 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $resolver->shouldStopOnViolation()
         );
 
+        $this->eventDispatcher->addListener(FixerFileProcessedEvent::NAME, [$progressOutput, 'onFixerFileProcessed']);
```

Comment:
why subscribing to event handling was taken outside of ProgressOutput?

Reply:
Because it does not make sense to inject event dispatcher to output writer. Previously dispatcher was injected there only for it to be invoked internally for attaching and detaching listener. It's not the writer's responsibility. Instead, `ProgressOutputInterface` defines a contract between the `FixCommand` and output writers.

Reply:
Also, as I said above, other output writers are added in #6842 and they also need to listen on that event, it does not make sense to inject event dispatcher to all writers and copy&paste attach/detach logic.

Reply:
fair point 👍🏻 
that reason was _not_ visible in commit log

Reply:
hmm, what I wonder here @Wirone .

Originally, listener was registered only if it was needed - now, it's registered even if `ProgressOutputType::NONE`